src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -461,23 +468,25 @@ private void EnsureMethodsImported()
             _helper.EnsureMethodsImported();
         }
 
-        public override void WriteXmlValue(XmlWriterDelegator xmlWriter, object obj, XmlObjectSerializerWriteContext context)
+        public override void WriteXmlValue(XmlWriterDelegator xmlWriter, object obj, XmlObjectSerializerWriteContext? context)
```

Comment:
most of the time when writing any objects they're nullable, please double check this should be non-null

Reply:
also when reading you get nullable

Reply:
At first I thought the same and I started to change this to `object? obj`, but I've looked both `WriteXmlValue` and `WriteJsonValueCore` over multiple times and everything that I can tell is that callers need to check for `null` and call `WriteNull` instead of calling this method. I assume that's what the "Value" in these methods mean - this value is valid.

I've confirmed that the major places that get called by these methods (primitive, enum, class and collection writers) will NullReferenceException or throw InvalidCastException if `obj` is `null`. So I believe these are correct - for these "WriteXXXValue" methods, `object obj` is not null.